### PR TITLE
Fix/complement groups information

### PR DESCRIPTION
**Pull request by bryanmg**
## Description
- Is required a field in groups to save the days that the group will take classes
- Is not possible know which users are members of a group

## Solution
- I added a new prop (days_of_the_day) in groups resource which will contain all days of the week and a flag is is an active day or not
- I added a relation between members (users) and groups

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -4,8 +4,8 @@ class User < ApplicationRecord
 
   enum :role, { teacher: "teacher", student: "student" }, scopes: false, default: "student"
   enum :belt, BELTS, scopes: false, default: nil
-
-  has_many :groups, dependent: :destroy
-
   validates :email, :name, :birth_date, presence: true
+
+  has_many :group, dependent: :destroy
```
Comment (itsmeurbi) [inline app/models/user.rb:9]: `has_many :group` ?

**Review by itsmeurbi — DISMISSED** [review]

File: app/models/user.rb (hunk 1 above)
Comment (bryanmg) [reply app/models/user.rb]: My bad, must be `has_many :groups`

File: app/models/user.rb [hunk 2]
```diff
@@ -4,8 +4,8 @@ class User < ApplicationRecord
 
   enum :role, { teacher: "teacher", student: "student" }, scopes: false, default: "student"
   enum :belt, BELTS, scopes: false, default: nil
+  validates :email, :name, :birth_date, presence: true
 
   has_many :groups, dependent: :destroy
-
-  validates :email, :name, :birth_date, presence: true
+  has_many :group_members, through: :group, dependent: :destroy
```
Comment (virgyjimenez) [inline app/models/user.rb:10]: `through: :group`  ?

Comment (itsmeurbi) [reply app/models/user.rb:10]: @bryanmg 👆🏻
I think from now on you should run a rails console and make sure relationships works as expected **before** `committing`

File: app/models/group.rb [hunk 3]
```diff
@@ -1,5 +1,9 @@
 class Group < ApplicationRecord
   belongs_to :user
+  store :days_of_the_week, accessors: [:sunday?, :monday?, :tuesday?, :wednesday?, :thursday?, :friday?, :saturday?]
 
   validates :from_time, :to_time, presence: true
+
+  has_many :group_members, dependent: :destroy
+  has_many :users, through: :group_members
```
Comment (bryanmg) [inline app/models/group.rb:8]: I tried to create a "members" relation but doesn't work for me, according to the documentation I tried with options like `:source, :as, and :class_name` but any of this doesn't work for me, so I decide leave like this and continues working.
examples that I tried:
`has_many :members, through: :group_members, class_name: "User"`
`has_many :members, through: :group_members, source: "User"`
`has_many :users, through: :group_members, as: "members"`

File: app/models/user.rb (hunk 2 above)
Comment (bryanmg) [reply app/models/user.rb]: @itsmeurbi I did, and It works well.
Fixed to `:groups` to improve code reading

**Review by virgyjimenez — APPROVED** [review]

**Pull request merged**